### PR TITLE
Release docs for terraform version V1.4.0

### PR DESCRIPTION
**Pull request by a-atri**
## Jira

* AV-87721

## Description
Release Documentation for Terraform version V1.4.0

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

### How was this change tested and do you have evidence? _**(REQUIRED: Select at least 1)**_

- [ ] Manually tested
- [ ] Unit tested
- [ ] Acceptance tested
- [x] Unable to test / will not test (Please provide comments in section below)

### Testing

<details open>
  <summary>Testing</summary>
  
</details>

## Required Checklist:

- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if required)
- [x] I have run make fmt and formatted my code
- [x] I have made sure that no schema field is marked with both requiresReplace and computed
## Further comments

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -298,3 +298,11 @@ To get started, see the [Provider Example Configs](https://github.com/couchbasec
 
   Manage Private Endpoints
 
+* [Manage Azure Network Peer](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/network_peer)
+
+  Manage Azure Network Peer
+
+* [Manage Flush Bucket](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/flush_bucket)
+
+  Manage Flush Bucket
```
Comment (simon-dew) [inline docs/index.md:307]: 1. There are two top-level headings in this document called **Example Usage**. To differentiate, rename this section **Links to Further Examples** or similar.

2. The top few entries in this list have more description under the link. The later entries, including these two, just repeat the link title. Could expand the descriptions for later entries including these two. Little point in just repeating the link title text.

3. The top few entries in this list have a colon (:) after the link. For consistency, either remove the colon from the top few entries or add a colon to all subsequent entries, including these two.

File: docs/guides/1.4.0-upgrade-guide.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Couchbase Capella Provider 1.4.0: Upgrade and Information Guide
+---
+
+
+# Couchbase Capella Provider 1.4.0: Upgrade and Information Guide
+
+New features in 1.4.0
+
+
+## New Features
```
Comment (simon-dew) [inline docs/guides/1.4.0-upgrade-guide.md:15]: Should this be after the heading?

```suggestion

## New Features

New features in 1.4.0:

```

File: docs/guides/1.4.0-upgrade-guide.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+* Enable Flush Bucket [`couchbase-capella_flush`](https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs/resources/flush)
+* Enable Azure Network Peer [`couchbase-capella_network_peer`](https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs/resources/network_peer)
+
+
+## Flush Bucket
+Use the new `flush` resource to flush a bucket
```
Comment (simon-dew) [inline docs/guides/1.4.0-upgrade-guide.md:21]: Full stop at end:

```suggestion
Use the new `flush` resource to flush a bucket.
```

**Review by simon-dew — CHANGES_REQUESTED** [review]
Quick scan of the documentation content. Couple of comments

File: docs/guides/1.4.0-upgrade-guide.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  project_id                     = var.project_id
+  cluster_id                     = var.cluster_id
+  tenant_id                      = var.tenant_id
+  vnet_id                        = var.vnet_id
+  subscription_id                = var.subscription_id
+  resource_group                 = var.resource_group
+  vnet_peering_service_principal = var.vnet_peering_service_principal
+}
+```
+Create a new Azure network peer, use the `network-peer` resource as follows:
```
Comment (simon-dew) [inline docs/guides/1.4.0-upgrade-guide.md:84]:
```suggestion
To create a new Azure network peer, use the `network-peer` resource as follows:
```

**Review by simon-dew — CHANGES_REQUESTED** [review]
Adding to previous comments

File: docs/guides/1.4.0-upgrade-guide.md (hunk 2 above)
Comment (rajagp) [reply docs/guides/1.4.0-upgrade-guide.md:15]: +1 to @simon-dew  suggestion

File: docs/index.md (hunk 1 above)
Comment (rajagp) [reply docs/index.md:307]: The description text needs to be fixed broadly. Started discussion in slack.

File: docs/guides/1.4.0-upgrade-guide.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    project_id      = var.project_id
+    cluster_id      = var.cluster_id
+    bucket_id       = var.bucket_id
+}
+```
+For more information, see the [flush bucket examples](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/flush_bucket).
+
+## Azure Network Peer
+
+Use the `azure_network_peer_command` data source to get an AWS command for configuring VPC endpoint.
```
Comment (rajagp) [inline docs/guides/1.4.0-upgrade-guide.md:66]:
```suggestion
Use the `azure_network_peer_command` data source to get an Azure CLI command for configuring VNET endpoint.
```

General: I have not paid close attention but please double check to ensure no copy paste errors. @PaulomeeCb  @a-atri

File: docs/guides/1.4.0-upgrade-guide.md (hunk 2 above)
Comment (a-atri) [reply docs/guides/1.4.0-upgrade-guide.md:15]: @simon-dew We can update it but in the past release we have done it like this only

New features in <Version Number> and in next line we have added "New Features" as heading text.

Attaching a screenshot of last release for reference. 
Screenshot 2024-11-25 at 1 00 10 PM

Comment (simon-dew) [reply docs/guides/1.4.0-upgrade-guide.md:15]: Fair enough! I'm happy to keep it as it is for consistency

File: docs/index.md
Comment (simon-dew) [inline docs/index.md:1]: 1. There are still two top-level headings in this file called **Example Usage**. Suggest changing the second one to **Links to Further Examples** or similar.
2. Glad to see there are further descriptions under the links. The last few links (from Manage Private Endpoints onward) still just repeat the link text.

**Review by simon-dew — COMMENTED** [review]
A couple of my comments from previous review still stand. I'm not requesting changes at this stage so as not to block the PR.

File: docs/index.md
Comment (simon-dew) [reply docs/index.md:1]: Feel free to dismiss this conversation and merge if you need to progress the PR

**Review by nishant-tripathy — DISMISSED** [review]
LGTM

File: docs/guides/1.4.0-upgrade-guide.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    project_id      = var.project_id
+    cluster_id      = var.cluster_id
+    bucket_id       = var.bucket_id
+}
+```
+For more information, see the [flush bucket examples](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/flush_bucket).
+
+## Azure Network Peer
+
+Use the `azure_network_peer_command` data source to get an Azure CLI command for configuring VNET endpoint.
```
Comment (PaulomeeCb) [inline docs/guides/1.4.0-upgrade-guide.md:66]: Can we add a line about the Admin consent granting process like-
```suggestion
Use the `azure_network_peer_command` data source to get an Azure CLI command for configuring VNET endpoint.

Note:  Before using this API, please make sure that the Admin consent granting process has been completed through the Capella UI. For more information, please refer to the [steps here] (https://docs.couchbase.com/cloud/management-api-reference/index.html#tag/Network-Peers/operation/getAzureVnetPeeringCommand)
```

**Pull request merged**